src/util/pg.py

```diff
@@ -1621,3 +1622,76 @@ def create_id_sequence(cr, table, set_as_default=True):
                 table=table_sql,
             )
         )
+
+
+def bulk_update_table(cr, table, columns, mapping, key_col="id"):
+    """
+    Update table based on mapping.
+
+    Each `mapping` entry defines the new values for the specified `columns` for the row(s)
+    whose `key_col` value matches the key.
+
+    .. example::
+
+       .. code-block:: python
+
+          # single column update
+          util.bulk_update_table(cr, "res_users", "active", {42: False, 27: True})
+
+          # multi-column update
+          util.bulk_update_table(
+              cr,
+              "res_users",
+              ["active", "login"],
+              {
+                  "administrator": [True, "admin"],
+                  "demouser": [True, "demo"],
+              },
+              key_col="login",
```

Comment:
Maybe, if we want an example of non-id key:
```suggestion
              ["active", "password"],
              {
                  "admin": [True, "admin"],
                  "demo": [True, "demo"],
              },
              key_col="login",
```

Reply:
I see. Although fine it's a bit confusing. In the example it's better to assume the existing value is what users get in a default install. The value set can then be different. 

IMO this blends too many options into a single example, and could be confusing. We should focus in the columns spec, that's the "weird" part of the function. I think it's fine to just put multi-column+explicit key column. Reflect this as well int the comment.

Nothing prevents updating the key column. Nothing implies that either from anything we said. IMO we don't need an example for it.